### PR TITLE
Infra/generify faded scroll view

### PR DESCRIPTION
**Pull request by M-i-k-e-l**
## Description
FadedScrollView - make more generic (I will use this in `Incubator.Dialog`, splitting up in to a few PRs).
Look at each commit separately for convenience.

## Changelog
None

### CODE REVIEW

File: src/components/tabController/FadedScrollView.tsx [hunk 1]
```diff
@@ -1,47 +1,80 @@
...
+  useGesture?: boolean;
+  children?: React.ReactNode | React.ReactNode[];
+};
 
 type Props = FadedScrollViewProps & ForwardRefInjectedProps;
-
-const FADER_SIZE = 76;
+interface Statics {
+  scrollTo(
+    y?: number | {x?: number | undefined; y?: number | undefined; animated?: boolean | undefined},
```
Comment (lidord-wix) [inline src/components/tabController/FadedScrollView.tsx]: Looks like the types are not in the right place here

File: src/components/tabController/FadedScrollView.tsx [hunk 2]
```diff
@@ -1,47 +1,80 @@
...
   } = useScrollReached({
-    horizontal: true,
+    horizontal,
     threshold: 50
   });
 
-  const showStart = scrollEnabled && !isScrollAtStart;
-  const showEnd = scrollEnabled && !isScrollAtEnd;
+  const showStart = (showStartFader && scrollEnabled && !isScrollAtStart) ?? false;
+  const showEnd = (showEndFader && scrollEnabled && !isScrollAtEnd) ?? false;
```
Comment (lidord-wix) [inline src/components/tabController/FadedScrollView.tsx]: Why do you need the `?? false` in these two lines?

**Review by lidord-wix — COMMENTED** [review]
Looks good, I wrote two small comments :)

File: src/components/tabController/FadedScrollView.tsx (hunk 1 above)
Comment (M-i-k-e-l) [reply src/components/tabController/FadedScrollView.tsx]: Not sure what you mean, can you elaborate \ rephrase?

File: src/components/tabController/FadedScrollView.tsx (hunk 2 above)
Comment (M-i-k-e-l) [reply src/components/tabController/FadedScrollView.tsx]: The default for the `Fader` is to be visible, I want it to not be visible unless it's needed.

File: src/components/tabController/FadedScrollView.tsx (hunk 1 above)
Comment (lidord-wix) [reply src/components/tabController/FadedScrollView.tsx]: Can the `y` key get both number and object?

**Review by lidord-wix — APPROVED** [review]

**Pull request merged**